src/components/ColorPickerInput.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  useState,
+  useEffect,
+  useRef,
+  MutableRefObject,
+  useLayoutEffect,
+} from "react";
+import { Box } from "@mui/material";
+
+import { fieldSx } from "@src/components/SideDrawer/utils";
+import { Color, ColorPicker } from "react-color-palette";
+import { useDebouncedCallback } from "use-debounce";
+
+const useResponsiveWidth = (): [
+  width: number,
+  setRef: MutableRefObject<HTMLElement | null>
+] => {
+  const ref = useRef(null);
+  const [width, setWidth] = useState(0);
+
+  useLayoutEffect(() => {
+    if (!ref || !ref.current) {
+      return;
+    }
+    const resizeObserver = new ResizeObserver((targets) => {
+      const { width: currentWidth } = targets[0].contentRect;
+      setWidth(currentWidth);
+    });
+
+    resizeObserver.observe(ref.current);
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, []);
+
+  return [width, ref];
+};
+
+export interface IColorPickerProps {
+  value: Color;
+  handleOnChangeComplete: (color: Color) => void;
+  disabled?: boolean;
+}
+
+export default function ColorPickerInput({
+  value,
+  handleOnChangeComplete,
+  disabled = false,
+}: IColorPickerProps) {
+  const [localValue, setLocalValue] = useState(value);
+  const [width, setRef] = useResponsiveWidth();
+
+  const debouncedOnChangeComplete = useDebouncedCallback((color) => {
+    handleOnChangeComplete(color);
+  }, 100);
+
+  useEffect(() => {
+    debouncedOnChangeComplete(localValue);
+  }, [debouncedOnChangeComplete, localValue]);
+
+  return (
+    <Box
+      ref={setRef}
+      sx={[
+        fieldSx,
+        {
+          marginTop: "1rem",
+          padding: "1rem",
+          borderColor: "divider",
+          transitionDuration: 0,
+          "& .rcp": {
+            border: "none",
+            "& .rcp-saturation": {
+              borderRadius: "4px",
```

Comment:
It'd be better to not use absolute units

Reply:
This is in MUI’s [`sx` prop](https://mui.com/material-ui/react-box/#the-sx-prop), so you can access our theme’s border-radius property with `borderRadius: 1`, which will set it to `1 * 4px`. You can find [all the properties here](https://mui.com/system/properties/).